relay-spans/src/otel_to_sentry_v2.rs

```diff
@@ -146,7 +146,43 @@ fn otel_value_to_attr(otel_value: OtelValue) -> Option<Attribute> {
             let s = String::from_utf8(bytes).ok()?;
             (AttributeType::String, Value::String(s))
         }
-        OtelValue::ArrayValue(_) | OtelValue::KvlistValue(_) => return None,
+        OtelValue::ArrayValue(array) => {
+            // Technically, `ArrayValue` can contain other arrays, or nested key-value lists. This
+            // is not usually allowed by the OTLP protocol, but for safety we filter those values
+            // out before serializing.
+            let safe_values: Vec<serde_json::Value> = array
+                .values
+                .into_iter()
+                .filter_map(|v| v.value)
+                .filter_map(|v| match v {
```

Comment:
As a general point: in `filter_map` you have access to `?`, meaning you can always collapse multiple `filter_map` calls.
```suggestion
                .filter_map(|v| match v.value?)
```
It's also fine to leave it IMO.

Reply:
I like this suggestion, I made the change, thanks!